pkg/handlers/index/delete.go

```diff
@@ -27,32 +28,36 @@ import (
 // Delete deletes a zinc index and its associated data. Be careful using thus as you ca't undo this action.
 
 // @Id DeleteIndex
-// @Summary Delete index
+// @Summary Delete indexes
 // @Tags    Index
 // @Produce json
-// @Param   index  path  string  true  "Index"
+// @Param   indexes  path  string  true  "indexes"
 // @Success 200 {object} meta.HTTPResponseIndex
 // @Failure 400 {object} meta.HTTPResponseError
 // @Failure 500 {object} meta.HTTPResponseError
-// @Router /api/index/{index} [delete]
+// @Router /api/index/{indexes} [delete]
 func Delete(c *gin.Context) {
-	indexName := c.Param("target")
-
-	index, exists := core.GetIndex(indexName)
-	if !exists {
-		c.JSON(http.StatusBadRequest, meta.HTTPResponseError{Error: "index " + indexName + " does not exists"})
+	indexNames := c.Param("target")
+	if indexNames == "" {
+		c.JSON(http.StatusBadRequest, meta.HTTPResponseError{Error: "index name cannot be empty"})
 		return
 	}
 
-	// delete meta
-	if err := core.DeleteIndex(index.Name); err != nil {
-		c.JSON(http.StatusInternalServerError, meta.HTTPResponseError{Error: err.Error()})
-		return
+	for _, indexName := range strings.Split(indexNames, ",") {
+		index, exists := core.GetIndex(indexName)
+		if !exists {
```

Comment:
As part of core.DeleteIndex we check if index exists , the check for index exists can be removed from line 47